mkosi

```diff
@@ -557,6 +557,26 @@ def mount_image(args, workspace, loopdev, root_dev, home_dev, srv_dev, root_read
 
             umount(root)
 
+@complete_step("Assigning hostname")
+def assign_hostname(args, workspace):
+    if not args.hostname:
+        return
+
+    root = os.path.join(workspace, "root")
+    hostname_path = os.path.join(root, "etc/hostname")
+    hosts_path = os.path.join(root, "etc/hosts")
+
+    with open(hostname_path, "w+") as f:
+        f.write("{}\n".format(args.hostname))
+
+    if os.path.isfile(hosts_path):
+        with open(hosts_path, "a") as f:
+            f.write("127.0.0.1\t{}\n".format(args.hostname))
+    else:
+        with open(hosts_path, "w+") as f:
+            f.write("127.0.0.1\tlocalhost\n")
+            f.write("127.0.0.1\t{}\n".format(args.hostname))
```

Comment:
There's no need to write `/etc/hosts`, `nss-myhostname` or `nss-resolved` cover this better.